pymepix/processing/rawfilesampler.py

```diff
@@ -72,10 +72,15 @@ def init_new_process(self, file):
             cent_timewalk_lut = np.load(self.cent_timewalk_file)
 
         self.packet_processor = PacketProcessor(start_time=self._startTime, timewalk_lut=timewalk_lut)
-        self.centroid_calculator = CentroidCalculator(**self._clustering_args,\
-                                                      cent_timewalk_lut=cent_timewalk_lut, \
-                                                      dbscan_clustering=self._dbscan_clustering,\
-                                                      number_of_processes=self._number_of_processes)
+        if self._clustering_args is not None:
+            self.centroid_calculator = CentroidCalculator(**self._clustering_args,\
```

Comment:
Can you please check, I believe if you put the **kwargs to the end of the function arguments, you don't need the if statement...

Reply:
Refactored